diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+        applyToCode(code){node ->
+            if(node.elementType == FUN)
+                Assert.assertFalse(node.isNodeFromCompanionObject())
+        }
+    }
+
+    @Test
+    fun `test isNodeFromFileLevel`() {
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if(node.treeParent != null && node.elementType == CLASS) {
+                Assert.assertTrue(node.isNodeFromFileLevel())
+            }
+        }
+    }
+
+    @Test
+    fun `test isValProperty`(){
+        val code = """
+            class Test() {
+            
+                private val name = "John"
+                
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var isVal = false
+        applyToCode(code) { node ->
+            if (node.isValProperty())
+                isVal = true
+        }
+        Assert.assertTrue(isVal)
+    }
+
+    @Test
+    fun `test isConst`(){
+        val code = """
+            class Test() {
+            
+                const val SPEED = 10
+                
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var isConst = false
+        applyToCode(code) { node ->
+            if (node.isConst())
+                isConst = true
+        }
+        Assert.assertTrue(isConst)
+    }
+
+    @Test
+    fun `test isVarProperty`(){
+        val code = """
+            class Test() {
+            
+                private var name: String? = null
+                
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var isVar = false
+        applyToCode(code) { node ->
+            if (node.isVarProperty())
+                isVar = true
+        }
+        Assert.assertTrue(isVar)
+    }
+
+    @Test
+    fun `test getAllLLeafsWithSpecificType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        val list = mutableListOf<ASTNode>()
+        var leafWithTypeCount = 0
+        var firstNode: ASTNode? = null
+        applyToCode(code){node ->
+            if(firstNode == null)
+                firstNode = node
+            if (node.isLeaf() && node.elementType == WHITE_SPACE) {
+                leafWithTypeCount++
```

Comment:
You can also collect nodes themselves to some list here and compare not only count of leaves, but entire lists in assert.